pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -459,11 +462,33 @@ private void handlePostCreation() throws Exception {
       }
     }
 
+    convertFormatIfNeeded(segmentOutputDir);
     LOGGER.info("Driver, record read time : {}", totalRecordReadTime);
     LOGGER.info("Driver, stats collector time : {}", totalStatsCollectorTime);
     LOGGER.info("Driver, indexing time : {}", totalIndexTime);
   }
 
+  // There are 3 options to correctly generate segments to v3 format
```

Comment:
This is only a variation of approach 2 - use mmap instead of holding data in memory. Updated comments to reflect that.